pkg/sentry/fsimpl/fuse/fusefs.go

```diff
@@ -150,8 +151,16 @@ func (fsType FilesystemType) GetFilesystem(ctx context.Context, vfsObj *vfs.Virt
 
 	fs.VFSFilesystem().Init(vfsObj, &fsType, fs)
 
-	// TODO: dispatch a FUSE_INIT request to the FUSE daemon server before
-	//  returning. Mount will not block on this dispatched request.
+	// dispatch a FUSE_INIT request to the FUSE daemon server before
+	// returning. Mount will not block on this dispatched request.
+	// Read the states of the kernel task right now since the
+	// resulting goroutine is different from the current one.
+	go func(k *kernel.Kernel, pid uint32) {
+		if err := fs.Init(creds, k, pid); err != nil {
+			log.Infof("FUSE_INIT failed: %v", err)
+			// TODO: Do something here. No clue what.
```

Comment:
Hmmm this is interesting. Yeah its unclear what we should do here, maybe unmount at this point? Maybe we surface a failed initialization error to all requests made to the filesystem after if this happens - asking the mounter to unmount and retry? @mrahatm @avagin may have better ideas